py_gtfs_rt_ingestion/py_gtfs_rt_ingestion/config_base.py

```diff
@@ -49,13 +60,48 @@ class ConfigDetail(ABC):
     ConfigDetail classes must implement all methods and properties that are
     defined.
     """
+
     @property
     @abstractmethod
-    def config_type(self) -> ConfigType: ...
+    def config_type(self) -> ConfigType:
+        """Get the config type for this config"""
 
     @property
     @abstractmethod
-    def export_schema(self) -> pyarrow.schema: ...
+    def export_schema(self) -> pyarrow.schema:
+        """Get the schema for the parquet table generated by this config"""
 
+    @property
     @abstractmethod
-    def record_from_entity(self, entity: dict) -> dict: ...
+    def transformation_schema(self) -> dict:
+        """
+        Get the transformation schema that describes how what json keys to copy
+        and how to remap them if necessary
+        """
```

Comment:
this used to be the unique portion of `record_from_entity`. move it into its on property method so it can be accessed by parent class.